base/locks.jl

```diff
@@ -1,33 +1,55 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-import Base: _uv_hook_close, unsafe_convert
+import Base: _uv_hook_close, unsafe_convert,
+    lock, trylock, unlock, islocked
 
-export SpinLock, Mutex,
-    init_lock!, destroy_lock!, lock!, trylock!, unlock!
+export SpinLock, RecursiveSpinLock, Mutex
 
 
 ##########################################
 # Atomic Locks
 ##########################################
 
+"""
+    AbstractLock
+
+Abstract supertype describing types that
+implement the thread-safe synchronization primitives:
+`lock`, `trylock`, `unlock`, and `islocked`
+"""
 abstract AbstractLock
 
 # Test-and-test-and-set spin locks are quickest up to about 30ish
 # contending threads. If you have more contention than that, perhaps
 # a lock is the wrong way to synchronize.
+"""
+    SpinLock()
```

Comment:
then the docstring should be for spinlock as written, and actually attached to the spinlock name instead of the tataslock name